mantidproject/mantid
# Make space for MeshObject

The geometry objects are rearranged to enable a MeshObject to be defined.

IObject class is defined. Object is renamed CGSObject and made to inherit from IObject.
ObjComponent is modified to use IObject instead of CSGObject.

**To test:**

Code Review and you may check that object can be rendered in instrument view or elsewhere.

Fixes #12827, except that MeshObject still needs to be added. The merger of this change beforehand will help with this.

CSGObject has a performance test. Run it (as the Object performance test) without this fix and with this fix to see whether there is a significant loss in performance.

Are you satisfied with the set of methods of IObject. Could any be removed?

**Release Notes** 
Will be added after MeshObject has been added.


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
